### PR TITLE
Add StringIO#close_read, closed_read?, close_write, closed_write?

### PR DESCRIPTION
**Pull request by MITSUBOSHI**
I added some methods in `string_io.rbs`.

- close_read
- closed_read?
- close_write
- closed_write?

### CODE REVIEW

File: stdlib/builtin/string_io.rbs [hunk 1]
```diff
@@ -6,4 +6,8 @@ class StringIO
   def write: (String) -> void
   def flush: () -> void
   def string: -> String
+  def close_read: -> NilClass
```
Comment (soutaro) [inline stdlib/builtin/string_io.rbs]: Could you use `nil` in this case?

There are some out-of-date `NilClass` uses in existing signatures and I'm sorry for confusion. We prefer writing `nil` usually.

Comment (MITSUBOSHI) [reply stdlib/builtin/string_io.rbs]: Thank you for reviewing this PR!!

Okay. I fixed it at https://github.com/ruby/ruby-signature/pull/71/commits/b6cdb488be324c239f7031877f927fea95032bdd .
Please let me know if commits in this PR should be squashed into one commit :)

Comment (soutaro) [reply stdlib/builtin/string_io.rbs]: @MITSUBOSHI Rebasing and squashing trivial fixes are preferred, if not too difficult! 👋

Comment (MITSUBOSHI) [reply stdlib/builtin/string_io.rbs]: @soutaro I did! Could you please review again?

Comment (soutaro) [reply stdlib/builtin/string_io.rbs]: Thanks!

**Pull request merged**